### PR TITLE
py-aioitertools: New package

### PR DESCRIPTION
**Pull request by a-saitoh-fj**
py-aioitertools: New package

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-aioitertools/package.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class PyAioitertools(Package):
+    """Implementation of itertools, builtins, and more for AsyncIO and mixed-type iterables."""
+
+    homepage = "https://aioitertools.omnilib.dev/en/stable/"
+    url      = "https://files.pythonhosted.org/packages/32/0b/3260ac050de07bf6e91871944583bb8598091da19155c34f7ef02244709c/aioitertools-0.7.1-py3-none-any.whl"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-aioitertools/package.py]: Can you switch to the source build?

File: var/spack/repos/builtin/packages/py-aioitertools/package.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+    version('0.7.1' , sha256='8972308474c41ed5e0636819f948ebff32f2318e70f7e7d23cd208c4357cc773', expand=False)
+
+    extends('python')
+    depends_on('python@3.6:3.7.999', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-aioitertools/package.py]:
```suggestion
    depends_on('python@3.6:', type=('build', 'run'))
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Comment by a-saitoh-fj** [comment]
Fixed the pointed out point.
- Fixed python dependencies.

And,
I'm sorry, I didn't have enough confirmation.
- The description was too long, so I put a line break
- I deleted it because there was extra space.

"Can you switch to the source build?"
I also tried a source build with tar.gz, but I couldn't build this oss because setup.py isn't available.

**Comment by adamjstewart** [comment]
If I download https://files.pythonhosted.org/packages/dc/a0/f149e54ea082e5f4ae71c68e8e72cab8749283329a7c2747336ef7b68dc0/aioitertools-0.7.1.tar.gz and extract it, there is a `setup.py` file right at the top level.

**Comment by a-saitoh-fj** [comment]
py-aioitertools: Switch to the source build

I'm sorry. The source code I was referring to was different.
I changed to the source build, so please check here.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**